src/Microsoft.VisualStudio.Extensibilty.Testing.SourceGenerator/TestServicesSourceGenerator.cs

```diff
@@ -0,0 +1,1014 @@
+    internal sealed class TestServiceAttribute : Attribute
+    {
+    }
+}
+";
+
+        private const string ErrorHandlerSource = @"// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.VisualStudio
+{
+    using System.Runtime.InteropServices;
+
+    internal static class ErrorHandler
+    {
+        public static bool Succeeded(int hr)
+            => hr >= 0;
+
+        public static bool Failed(int hr)
+            => hr < 0;
+
+        public static int ThrowOnFailure(int hr)
+        {
+            if (Failed(hr))
+            {
+                Marshal.ThrowExceptionForHR(hr);
+            }
+
+            return hr;
+        }
+    }
+}
+";
+
+        private const string VSConstantsSource = @"// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.VisualStudio
+{
+    using System;
+
+    internal static partial class VSConstants
+    {
+        // General HRESULTS
+
+        /// <summary>HRESULT for FALSE (not an error).</summary>
+        public const int S_FALSE = 0x00000001;
+        /// <summary>HRESULT for generic success.</summary>
+        public const int S_OK = 0x00000000;
+
+        /// <summary>
+        /// These element IDs are the only element IDs that can be used with the selection service.
+        /// </summary>
+        public enum VSSELELEMID
+        {
+            SEID_UndoManager = 0,
+            SEID_WindowFrame = 1,
+            SEID_DocumentFrame = 2,
+            SEID_StartupProject = 3,
+            SEID_PropertyBrowserSID = 4,
+            SEID_UserContext = 5,
+            SEID_ResultList = 6,
+            SEID_LastWindowFrame = 7,
+        }
+    }
+}
+";
+
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            context.RegisterPostInitializationOutput(context =>
+            {
+                context.AddSource($"SolutionExplorerInProcess1{SourceSuffix}", SolutionExplorerInProcessSource);
+                context.AddSource($"ShellInProcess1{SourceSuffix}", ShellInProcessSource);
+                context.AddSource($"TestServiceAttribute{SourceSuffix}", TestServiceAttributeSource);
+            });
+
+            var services = context.SyntaxProvider.CreateSyntaxProvider(
+                predicate: static (node, cancellationToken) =>
+                {
+                    if (node is AttributeSyntax attribute)
+                    {
+                        var unqualifiedName = GetUnqualifiedName(attribute.Name).Identifier.ValueText;
+                        return unqualifiedName is "TestService" or "TestServiceAttribute";
+                    }
+
+                    return false;
+                },
+                static (context, cancellationToken) =>
```

Comment:
Nit: Maybe provide the parameter name for this?

Reply:
Will include this with the follow-up